packages/dd-trace/src/appsec/index.js

```diff
@@ -192,10 +195,26 @@ function onRequestQueryParsed (channelData) {
   checkRequestData(channelData, getQueryPayload(channelData.req))
 }
 
+function hasPathParamsChanged (store, params) {
+  if (store) {
+    if (JSON.stringify(store[BLOCKING_PATH_PARAMS_KEY]) !== JSON.stringify(params)) {
+      store[BLOCKING_PATH_PARAMS_KEY] = params
+      return true
+    } else {
+      return false
+    }
+  }
+  // REVIEW: If store can't be accesed better to return true?
+  return true
+}
+
 function getPathParamsPayload (req) {
   if (req.params && typeof req.params === 'object') {
-    return {
-      [addresses.HTTP_INCOMING_PARAMS]: req.params
+    const store = storage.getStore()
```

Comment:
`storage.getStore()` could be moved to `hasPathParamsChanged` and leave only one conditional here:

`if (req.params && typeof req.params === 'object' && hasPathParamsChanged(req.params))`